app/models/manageiq/providers/azure/network_manager/refresh_parser.rb

```diff
@@ -93,6 +102,128 @@ def get_floating_ips
     process_collection(floating_ips, :floating_ips) { |n| parse_floating_ip(n) }
   end
 
+  def get_load_balancers
+    process_collection(load_balancers, :load_balancers) { |lb| parse_load_balancer(lb) }
+  end
+
+  def get_load_balancer_pools
+    @data[:load_balancer_pool_members] = []
+
+    load_balancers.each do |lb|
+      process_collection(lb.properties["backendAddressPools"], :load_balancer_pools) do |pool|
+        parse_load_balancer_pool(pool)
+      end
+    end
+  end
+
+  def get_load_balancer_listeners
+    load_balancers.each do |lb|
+      process_collection(lb.properties["loadBalancingRules"], :load_balancer_listeners) do |listener|
+        parse_load_balancer_listener(lb, listener)
+      end
+    end
+  end
+
+  def get_load_balancer_health_checks
+    load_balancers.each do |lb|
+      process_collection(lb.properties["probes"], :load_balancer_health_checks) do |health_check|
+        parse_load_balancer_health_check(lb, health_check)
+      end
+    end
+  end
+
+  def get_heath_check_listener_and_members(health_check)
+    health_check_members   = []
+    matched_listener       = nil
+
+    health_check.properties["loadBalancingRules"].to_a.each do |health_check_listener|
+      matched_listener = @data.fetch_path(:load_balancer_listeners).detect do |listener|
+        listener[:ems_ref] == health_check_listener["id"]
```

Comment:
@ladas -When searching for a matched listener in AWS the port and protocol must match in addition to the UID. In Azure, the listener used by a health check does not use a matching protocol, one uses TCP and other other uses HTTP. Therefore, I dropped the protocol comparison here.